ESP8266 Beacon Client/BeaconFader.cpp

```diff
@@ -0,0 +1,59 @@
+#include "BeaconFader.h"
+#include "BeaconColor.h"
+
+BeaconFader::BeaconFader() {
+  init(BeaconColor::off(), BeaconColor::off());  
+}
+
+BeaconFader::BeaconFader(BeaconColor sourceColor, BeaconColor destinationColor) {
+  init(sourceColor, destinationColor);
+}
+
+void BeaconFader::init(BeaconColor sourceColor, BeaconColor destinationColor) {
+  this -> sourceColor = sourceColor;
+  this -> destinationColor = destinationColor;
+  red = sourceColor.getRed();
+  green = sourceColor.getGreen();
+  blue = sourceColor.getBlue();
+  white = sourceColor.getWhite();
+  rRatio = ((double)(sourceColor.getRed() - destinationColor.getRed())/((double)255)) * -1;
+  gRatio = ((double)(sourceColor.getGreen() - destinationColor.getGreen())/((double)255)) * -1;
+  bRatio = ((double)(sourceColor.getBlue() - destinationColor.getBlue())/((double)255)) * -1;
+  wRatio = ((double)(sourceColor.getWhite() - destinationColor.getWhite())/((double)255)) * -1;
+}
+
+bool BeaconFader::isComplete() {
+  return getCurrentColor().equals(destinationColor);
+}
+
+BeaconColor BeaconFader::getNext() {
+  red = red + rRatio;
```

Comment:
Could use combination operator.
```suggestion
  red += rRatio;
```